sim/pkg/model/metrics_entity.go

```diff
@@ -0,0 +1,45 @@
+package model
+
+import (
+	"fmt"
+	"github.com/josephburnett/sk-plugin/pkg/skplug/proto"
+	"skenario/pkg/simulator"
+)
+
+type Metrics interface {
+	GetStats() []*proto.Stat
+}
+
+type MetricsEntity interface {
+	simulator.Entity
+	Metrics
+}
+
+type metricsEntity struct {
+	env    simulator.Environment
```

Comment:
Why do you need to store the environment?  Isn't this stock already embedded in the environment where it needs to be?